src/pages/career/[opportunityId].tsx

```diff
@@ -1,19 +1,35 @@
-import { useRouter } from 'next/router';
+import { NextPage, NextPageContext } from 'next';
 import React from 'react';
 
 import DetailView from 'career/containers/DetailView';
-import CareerOpportunities from 'career/providers/CareerProvider';
 
-const CareerDetailPage = () => {
-  const router = useRouter();
-  const opportunityId = Number(router.query.opportunityId);
+import { careerOpportunitySelectors, fetchCareerOpportunityById } from 'career/slices/careerOpportunities';
+import { Store } from 'core/redux/Store';
+
+interface IContext extends NextPageContext {
+  store: Store;
+}
+
+interface IProps {
+  opportunityId: number;
+}
+
+const CareerDetailPage: NextPage<IProps> = ({ opportunityId }) => {
   return (
     <section>
-      <CareerOpportunities>
-        <DetailView opportunityId={opportunityId} />
-      </CareerOpportunities>
+      <DetailView opportunityId={opportunityId} />
     </section>
   );
 };
 
+CareerDetailPage.getInitialProps = async ({ query, store }: IContext) => {
+  const opportunityId = Number(query.opportunityId);
+  const isOpportunityInStore = Boolean(careerOpportunitySelectors.selectById(store.getState(), opportunityId));
+  const result = store.dispatch(fetchCareerOpportunityById(opportunityId));
```

Comment:
Is there a difference in getting store as a prop and using `store.dispatch` vs. importing `useDispatch()`?

Reply:
There is a difference here, since `Page.getInitialProps` is part of the Next runtime, not React itself.
`useDispatch` will get the dispatch function from the current store through the redux context from the redux provider.
Since `getInitialProps` is run outside of the React component tree there is no provider/context to the the current store/dispatch function from.

To allow our `getInitialProps` function to access the redux store and dispatch functions the store is passed to all `getInitialProps` functions from the `_app` component, which is where the store is created (or retrieved if it already exists). This is done under the hood by `next-redux-wrapper`

Reply:
I see! Neat stuff!